testing/integration/otel_helm_test.go

```diff
@@ -0,0 +1,189 @@
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/cli/values"
+	"helm.sh/helm/v3/pkg/getter"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+)
+
+var (
+	kubeStackChartURL     = "https://github.com/open-telemetry/opentelemetry-helm-charts/releases/download/opentelemetry-kube-stack-0.3.2/opentelemetry-kube-stack-0.3.2.tgz"
+	kubeStackChartVersion = "0.3.2"
+)
+
+func TestOtelKubeStackHelm(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Stack: &define.Stack{},
+		Local: false,
+		Sudo:  false,
+		OS: []define.OS{
+			// only test the basic and the wolfi container with otel
+			{Type: define.Kubernetes, DockerVariant: "basic"},
+			{Type: define.Kubernetes, DockerVariant: "wolfi"},
+		},
+		Group: define.Kubernetes,
+	})
+
+	agentImage := os.Getenv("AGENT_IMAGE")
+	require.NotEmpty(t, agentImage, "AGENT_IMAGE must be set")
+
+	agentImageParts := strings.SplitN(agentImage, ":", 2)
+	require.Len(t, agentImageParts, 2, "AGENT_IMAGE must be in the form '<repository>:<version>'")
+	agentImageRepo := agentImageParts[0]
+	agentImageTag := agentImageParts[1]
+
+	client, err := info.KubeClient()
+	require.NoError(t, err)
+	require.NotNil(t, client)
+
+	testLogsBasePath := os.Getenv("K8S_TESTS_POD_LOGS_BASE")
+	require.NotEmpty(t, testLogsBasePath, "K8S_TESTS_POD_LOGS_BASE must be set")
+
+	err = os.MkdirAll(filepath.Join(testLogsBasePath, t.Name()), 0o755)
+	require.NoError(t, err, "failed to create test logs directory")
+
+	namespace := info.Namespace
+
+	esHost := os.Getenv("ELASTICSEARCH_HOST")
+	require.NotEmpty(t, esHost, "ELASTICSEARCH_HOST must be set")
+
+	esAPIKey, err := generateESAPIKey(info.ESClient, namespace)
+	require.NoError(t, err, "failed to generate ES API key")
+	require.NotEmpty(t, esAPIKey, "failed to generate ES API key")
+
+	chartOptions := &action.ChartPathOptions{
+		RepoURL: "https://github.com/open-telemetry/opentelemetry-helm-charts/releases/download/opentelemetry-kube-stack-0.3.2/opentelemetry-kube-stack-0.3.2.tgz",
+		Version: "0.3.0",
+	}
+
+	chartLocation, err := action.NewPull().LocateChart(chartOptions.RepoURL, cli.New())
+	if err != nil {
+		panic(err)
+	}
+
+	testCases := []struct {
+		name                       string
+		helmReleaseName            string
+		valuesFile                 string
+		atLeastValidatedPodsNumber int
+	}{
+		{
+			name:            "helm standalone agent default kubernetes privileged",
+			helmReleaseName: "kube-stack-otel",
+			valuesFile:      "../../deploy/helm/edot-collector/kube-stack/values.yaml",
+			// - perNode Daemonset (at least 1 agent pod)
+			// - clusterWide Deployment  (1 agent pod)
+			// - operator Deployment  (1 agent pod)
```

Comment:
nit: irrelevant comments, right?!

Reply:
I think it can help a non-familiar user to know what the "kube-stack" Helm chart will deploy without taking a deep look into the values.yaml file.

Nonetheless, I am fine removing them or referencing another documentation source.

Reply:
sorry if I fail to see it, but what `perNode` and `clusterWide` mean in the context of EDOT collector?